### ISSUE TITLE
Zarf package create failed for Git components with no extra branches/tags

### ISSUE THREAD

**Issue by alita1991** [defenseunicorns/zarf#1982]
### Environment
Device and OS: Ubuntu
App version: v0.28.2
Kubernetes distro being used: k3s

### Steps to reproduce
1. Create a new repository in GitLab
2. Define in the _zarf.yaml_ file the Git component
3. Execute zarf package create

### Expected result
A package should be created

### Actual Result
The package is not created

### Visual Proof (screenshots, videos, text, etc)
unable to add component: unable to pull git repo _GIT_HTTPS_URL_, not a valid git repo or unable to clone (_GIT_HTTPS_URL_): already up-to-date

### Severity/Priority
Medium

### Additional Context
I looked into the code and found out that this is not working only for cases where I have no additional branches/tags, here is the code:

```
// Clone the git repository. 
err = g.clone(gitURLNoRef, ref, shallow) // <--- this works well
if err != nil {
	return fmt.Errorf("not a valid git repo or unable to clone (%s): %w", gitURL, err)
}


// If we're cloning the whole repo, we need to also fetch the other branches besides the default.
if ref == emptyRef { // <-- ref is equal to emptyRef
	fetchOpts := &git.FetchOptions{
		RemoteName: onlineRemoteName,
		Progress:   g.Spinner,
		RefSpecs:   []goConfig.RefSpec{"refs/*:refs/*"},
		Tags:       git.AllTags,
	}

	if gitCred != nil {
		fetchOpts.Auth = &gitCred.Auth
	}

	if err := repo.Fetch(fetchOpts); err != nil { // <-- fails here with already up-to-date
		return err
	}
}
```

**Comment by Noxsios** [comment]
@alita1991 We have seen similar stuff in the past, a current workaround is noted in this comment: https://github.com/defenseunicorns/zarf/issues/1939#issuecomment-1662216970 , but this behavior should have been fixed, thanks for the issue.

**Comment by Racer159** [comment]
@alita1991 was that able to resolve your issue?

**Comment by benmountjoy111** [comment]
FYI, i saw this on v0.28.2 as well, updated zarf to v0.29.0 and it fixed issue. So i think this could probably be closed as the latest version as of today (v0.29.0) does not experience the above error anymore.

**Comment by Racer159** [comment]
Per @benmountjoy111 's comment going to close this issue - can reopen if it comes back
